### DISCUSSION TITLE
Horizontal scrolling beyond bounds

### DISCUSSION THREAD

**Discussion by Zineffable1** [sumatrapdfreader/sumatrapdf#2463] [Q&A]
Is it possible?
![suma1](https://user-images.githubusercontent.com/61528972/162101300-87afab39-8a4c-405d-aacf-4ee89a81f0ec.png)
So there I am reading like normal.
![suma2](https://user-images.githubusercontent.com/61528972/162101394-15af3274-4840-4984-a1d3-c088bf4121e7.png)
And here I have zoomed out well beyond the bounds.

Is there a way to scroll horizontally beyond the bounds in the first image? i.e. centre the text.

**Comment by kjk** [comment]
Unfortunately no, this is not possible.

**Reply by Zineffable1** [reply]
How would one go about implementing this?

**Reply by GitHubRulesOK** [reply]
Not sure there is anything to implement for your example

That text is set as a broad right hand column thus to be centered it needs to be changed in the source editor, the alternative is to zoom in so that text is predominantly as wide as the screen then you can scroll it rightwards but not left due to the crop/media boundary thus in order to scan media to the right the crop/media boundary would need to have been set with a larger X value in source or using an editor.

**Reply by kjk** [reply]
It's probably not impossible but also not easy as it violates how scrollbars on Windows work. Most of the scrolling logic is in SumatraPDF.cpp (see WM_VSCROLL, WM_HSCROLL) but that's all I can really say. I don't see an obvious way to implement that.

**Comment by Zineffable1** [comment]
ok, thanks yall
